panther_power_control/src/relays_node.py

```diff
@@ -73,37 +76,42 @@ def __init__(self, name: str) -> None:
         rospy.loginfo(f'[{rospy.get_name()}] Node started')
 
     def _cmd_vel_cb(self, *args) -> None:
-        self._cmd_vel_msg_time = rospy.get_time()
+        with self._lock:
+            self._cmd_vel_msg_time = rospy.get_time()
 
     def _motor_controllers_state_cb(self, msg: DriverState) -> None:
-        self._can_net_err = any({msg.rear.fault_flag.can_net_err, msg.front.fault_flag.can_net_err})
+        with self._lock:
+            self._can_net_err = any({msg.rear.fault_flag.can_net_err, msg.front.fault_flag.can_net_err})
 
     def _e_stop_reset_cb(self, req: TriggerRequest) -> TriggerResponse:
-        if rospy.get_time() - self._cmd_vel_msg_time <= 2.0:
-            return TriggerResponse(
-                False,
-                'E-STOP reset failed, messages are still published on /cmd_vel topic!',
-            )
-        elif self._can_net_err:
-            return TriggerResponse(
-                False,
-                'E-STOP reset failed, unable to communicate with motor controllers! Please check connection with motor controllers.',
-            )
-        
-        self._e_stop_state = False
-        self._e_stop_state_pub.publish(self._e_stop_state)
-        return TriggerResponse(True, 'E-STOP reset successful')
+        with self._lock:
+            if rospy.get_time() - self._cmd_vel_msg_time <= 2.0:
+                return TriggerResponse(
+                    False,
+                    'E-STOP reset failed, messages are still published on /cmd_vel topic!',
+                )
+            elif self._can_net_err:
+                return TriggerResponse(
+                    False,
+                    'E-STOP reset failed, unable to communicate with motor controllers! Please check connection with motor controllers.',
+                )
+            
+            self._e_stop_state = False
+            self._e_stop_state_pub.publish(self._e_stop_state)
+            return TriggerResponse(True, 'E-STOP reset successful')
 
     def _e_stop_trigger_cb(self, req: TriggerRequest) -> TriggerResponse:
-        self._e_stop_state = True
-        self._e_stop_state_pub.publish(self._e_stop_state)
-        return TriggerResponse(True, 'E-SROP triggered successful')
+        with self._lock:
+            self._e_stop_state = True
+            self._e_stop_state_pub.publish(self._e_stop_state)
+            return TriggerResponse(True, 'E-SROP triggered successful')
 
     def _set_motor_state_timer_cb(self, *args) -> None:
-        motor_state = GPIO.input(self._pins.STAGE2_INPUT)
-        GPIO.output(self._pins.MOTOR_ON, motor_state)
-        if motor_state != self._motor_on_pub.impl.latch.data:
-            self._motor_on_pub.publish(motor_state)
+        with self._lock:
```

Comment:
You need to consider what should be included in methods like this, as the implementation of `_e_stop_event` method in `power_board_node.py` differs.